@kenfinnigan I updated to use version 1.2.1 of the TCK, but there were still failures when I ran it locally.

For example here is one:

`Caused by: java.util.NoSuchElementException: Property org.eclipse.microprofile.rest.client.tck.interfaces.SimpleGetApi/mp-rest/url not found`

--- a/build-parent/pom.xml
+++ b/build-parent/pom.xml
@@ -63,6 +63,7 @@
         <!-- MicroProfile TCK versions -->
         <microprofile-health-api.version>1.0</microprofile-health-api.version>
         <microprofile-config-api.version>1.3</microprofile-config-api.version>
+        <microprofile-rest-client-api.version>1.2.1</microprofile-rest-client-api.version>
     </properties>
 
     <dependencyManagement>

--- a/extensions/smallrye-rest-client/runtime/src/main/java/io/quarkus/smallrye/restclient/runtime/RestClientBase.java
+++ b/extensions/smallrye-rest-client/runtime/src/main/java/io/quarkus/smallrye/restclient/runtime/RestClientBase.java
@@ -18,6 +18,7 @@ package io.quarkus.smallrye.restclient.runtime;
 
 import java.net.MalformedURLException;
 import java.net.URL;
+import java.util.Optional;
 
 import org.eclipse.microprofile.config.Config;
 import org.eclipse.microprofile.config.ConfigProvider;
@@ -25,7 +26,9 @@ import org.eclipse.microprofile.rest.client.RestClientBuilder;
 
 public class RestClientBase {
 
-    public static final String REST_URL_FORMAT = "%s/mp-rest/url";
+    private static final String MP_REST = "mp-rest";
+    public static final String REST_URL_FORMAT = "%s/" + MP_REST + "/url";
+    public static final String REST_URI_FORMAT = "%s/" + MP_REST + "/uri";
 
     private final Class<?> proxyType;
     private final String baseUriFromAnnotation;
@@ -55,10 +58,19 @@ public class RestClientBase {
     }
 
     private String getBaseUrl() {
-        if ((baseUriFromAnnotation == null) || baseUriFromAnnotation.isEmpty()) {
-            String property = String.format(REST_URL_FORMAT, proxyType.getName());
-            return config.getValue(property, String.class);
+        String propertyName = String.format(REST_URI_FORMAT, proxyType.getName());
+        Optional<String> propertyOptional = config.getOptionalValue(propertyName, String.class);
+        if (!propertyOptional.isPresent()) {
+            propertyName = String.format(REST_URL_FORMAT, proxyType.getName());
+            propertyOptional = config.getOptionalValue(propertyName, String.class);
         }
-        return baseUriFromAnnotation;
+        if (((baseUriFromAnnotation == null) || baseUriFromAnnotation.isEmpty())
+                && !propertyOptional.isPresent()) {
+            throw new IllegalArgumentException(
+                    String.format(
+                            "Unable to determine the proper baseUrl/baseUri. Consider registering using @RegisterRestClient(baseUri=\"someuri\"), or by adding '%s' to your Quarkus configuration",
+                            propertyName));
+        }
+        return propertyOptional.orElse(baseUriFromAnnotation);
     }
 }

--- a/tcks/microprofile-rest-client/pom.xml
+++ b/tcks/microprofile-rest-client/pom.xml
@@ -1,0 +1,81 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-tck-parent</artifactId>
+        <version>999-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-tck-microprofile-rest-client</artifactId>
+    <name>Quarkus - TCK - MicroProfile REST Client</name>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <suiteXmlFiles>
+                        <suiteXmlFile>tck-suite.xml</suiteXmlFile>
+                    </suiteXmlFiles>
+                    <systemPropertyVariables>
+                        <!-- Disable quarkus optimization -->
+                        <quarkus.arc.remove-unused-beans>false</quarkus.arc.remove-unused-beans>
+                    </systemPropertyVariables>
+                    <!-- This workaround allows us to run a single test using
+                        the "test" system property -->
+                    <!-- https://issues.apache.org/jira/browse/SUREFIRE-569 -->
+                    <dependenciesToScan>
+                        <dependency>org.eclipse.microprofile.rest.client:microprofile-rest-client-tck</dependency>
+                    </dependenciesToScan>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>uk.co.deliverymind</groupId>
+                <artifactId>wiremock-maven-plugin</artifactId>
+                <version>2.7.0</version>
+                <executions>
+                    <execution>
+                        <phase>generate-test-sources</phase>
+                        <goals>
+                            <goal>run</goal>
+                        </goals>
+                        <configuration>
+                            <dir>target/classes</dir>
+                            <params>--port=8765 --verbose</params>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-arquillian</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-smallrye-rest-client</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.microprofile.rest.client</groupId>
+            <artifactId>microprofile-rest-client-tck</artifactId>
+            <version>${microprofile-rest-client-api.version}</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.eclipse.jetty</groupId>
+                    <artifactId>jetty-servlet</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.eclipse.jetty</groupId>
+                    <artifactId>jetty-servlets</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+    </dependencies>
+</project>

--- a/tcks/microprofile-rest-client/src/test/java/io/quarkus/tck/restclient/SkipTestsThatThrowNoClassDefFound.java
+++ b/tcks/microprofile-rest-client/src/test/java/io/quarkus/tck/restclient/SkipTestsThatThrowNoClassDefFound.java
@@ -1,0 +1,44 @@
+package io.quarkus.tck.restclient;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.microprofile.rest.client.tck.ClientHeaderParamTest;
+import org.eclipse.microprofile.rest.client.tck.InvokeWithJsonBProviderTest;
+import org.eclipse.microprofile.rest.client.tck.InvokeWithJsonPProviderTest;
+import org.testng.IAnnotationTransformer;
+import org.testng.IConfigurable;
+import org.testng.IConfigureCallBack;
+import org.testng.ITestNGMethod;
+import org.testng.ITestResult;
+import org.testng.annotations.ITestAnnotation;
+
+public class SkipTestsThatThrowNoClassDefFound implements IConfigurable, IAnnotationTransformer {
+
+    private static final Set<String> TESTS_THAT_THROW_CLASS_NOT_FOUND = new HashSet<>(Arrays.asList(
+            InvokeWithJsonBProviderTest.class.getName(), InvokeWithJsonPProviderTest.class.getName(),
+            ClientHeaderParamTest.class.getName()));
+
+    // ensures that the methods annotated with @BeforeTest don't run (since this is where the exception is thrown)
+    @Override
+    public void run(IConfigureCallBack callBack, ITestResult testResult) {
+        ITestNGMethod method = testResult.getMethod();
+        if (TESTS_THAT_THROW_CLASS_NOT_FOUND.contains(method.getTestClass().getName())) {
+            return;
+        }
+
+        callBack.runConfigurationMethod(testResult);
+    }
+
+    // ensures the actual tests method don't run
+    @Override
+    public void transform(ITestAnnotation annotation, Class testClass, Constructor testConstructor, Method testMethod) {
+        String className = testClass != null ? testClass.getName() : testMethod.getDeclaringClass().getName();
+        if (TESTS_THAT_THROW_CLASS_NOT_FOUND.contains(className)) {
+            annotation.setEnabled(false);
+        }
+    }
+}

--- a/tcks/microprofile-rest-client/tck-suite.xml
+++ b/tcks/microprofile-rest-client/tck-suite.xml
@@ -1,0 +1,129 @@
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+<suite name="microprofile-rest-client-TCK" verbose="2"
+    configfailurepolicy="continue">
+
+    <!--
+        There are classes in DisableTestsTransformer that throw:
+        java.lang.NoClassDefFoundError: org/jboss/resteasy/specimpl/ResteasyUriBuilder
+        These need to be disabled completely because simply ignoring all methods is not good enough
+        as the exception occurs during @BeforeTest
+    -->
+    <listeners>
+        <listener class-name="io.quarkus.tck.restclient.SkipTestsThatThrowNoClassDefFound" />
+    </listeners>
+
+    <test name="microprofile-rest-client 1.2 TCK">
+        <packages>
+            <package
+                name="org.eclipse.microprofile.rest.client.tck.*">
+                <!-- timeout not implemented -->
+                <exclude name="org.eclipse.microprofile.rest.client.tck.timeout"/>
+            </package>
+        </packages>
+        
+        <classes>
+            <!-- mp/scope not implemented -->
+            <class name="org.eclipse.microprofile.rest.client.tck.cditests.HasAppScopeTest">
+                <methods>
+                    <exclude name=".*" />
+                </methods>
+            </class>
+            <class name="org.eclipse.microprofile.rest.client.tck.cditests.HasConversationScopeTest">
+                <methods>
+                    <exclude name=".*" />
+                </methods>
+            </class>
+            <class name="org.eclipse.microprofile.rest.client.tck.cditests.HasRequestScopeTest">
+                <methods>
+                    <exclude name=".*" />
+                </methods>
+            </class>
+            <class name="org.eclipse.microprofile.rest.client.tck.cditests.HasSessionScopeTest">
+                <methods>
+                    <exclude name=".*" />
+                </methods>
+            </class>
+
+            <!-- scope is singleton by default -->
+            <class name="org.eclipse.microprofile.rest.client.tck.cditests.CDIInvokeSimpleGetOperationTest">
+                <methods>
+                    <exclude name="testHasDependentScopedByDefault" />
+                </methods>
+            </class>
+            <class name="org.eclipse.microprofile.rest.client.tck.asynctests.CDIInvokeAsyncSimpleGetOperationTest">
+                <methods>
+                    <exclude name="testHasDependentScopedByDefault" />
+                </methods>
+            </class>
+
+            <!-- can't inject without @RestClient qualifier -->
+            <class name="org.eclipse.microprofile.rest.client.tck.InvokeWithJsonPProviderTest">
+                <methods>
+                    <exclude name=".*" />
+                </methods>
+            </class>
+            <class name="org.eclipse.microprofile.rest.client.tck.FeatureRegistrationTest">
+                <methods>
+                    <exclude name=".*" />
+                </methods>
+            </class>
+
+            <!-- RESTEASY004560: Failed to buffer aborted response -->
+            <class name="org.eclipse.microprofile.rest.client.tck.CustomHttpMethodTest">
+                <methods>
+                    <exclude name=".*" />
+                </methods>
+            </class>
+            <class name="org.eclipse.microprofile.rest.client.tck.RestClientBuilderListenerTest">
+                <methods>
+                    <exclude name=".*" />
+                </methods>
+            </class>
+            <class name="org.eclipse.microprofile.rest.client.tck.RestClientListenerTest">
+                <methods>
+                    <exclude name=".*" />
+                </methods>
+            </class>
+
+            <!-- RESTEASY003145: Unable to find a MessageBodyReader -->
+            <class name="org.eclipse.microprofile.rest.client.tck.cditests.CDIInterceptorTest">
+                <methods>
+                    <exclude name=".*" />
+                </methods>
+            </class>
+            <class name="org.eclipse.microprofile.rest.client.tck.ClientHeadersFactoryTest">
+                <methods>
+                    <exclude name=".*" />
+                </methods>
+            </class>
+            <class name="org.eclipse.microprofile.rest.client.tck.ClientHeaderParamTest">
+                <methods>
+                    <exclude name=".*" />
+                </methods>
+            </class>
+            <class name="org.eclipse.microprofile.rest.client.tck.ClientReuseTest">
+                <methods>
+                    <exclude name=".*" />
+                </methods>
+            </class>
+
+
+            <!-- providers related -->
+            <class name="org.eclipse.microprofile.rest.client.tck.InvokeWithBuiltProvidersTest">
+                <methods>
+                    <exclude name="testInvokesPutOperationWithRegisteredProviders" />
+                    <exclude name="testInvokesPostOperationWithRegisteredProviders" />
+                </methods>
+            </class>
+            <class name="org.eclipse.microprofile.rest.client.tck.cditests.CDIInvokeWithRegisteredProvidersTest">
+                <methods>
+                    <exclude name="testInvokesPostOperation_viaMPConfig" />
+                    <exclude name="testInvokesPutOperation_viaMPConfig" />
+                </methods>
+            </class>
+
+        </classes>
+
+    </test>
+
+</suite>

--- a/tcks/pom.xml
+++ b/tcks/pom.xml
@@ -16,6 +16,7 @@
     <modules>
         <module>microprofile-config</module>
         <module>microprofile-health</module>
+        <module>microprofile-rest-client</module>
     </modules>
 
     <properties>

